app/javascript/packs/flagUserModal.js

```diff
@@ -0,0 +1,158 @@
+          <path d="M12 10.586l4.95-4.95 1.414 1.414-4.95 4.95 4.95 4.95-1.414 1.414-4.95-4.95-4.95 4.95-1.414-1.414 4.95-4.95-4.95-4.95L7.05 5.636l4.95 4.95z" />
+        </svg>
+      </button>
+    </header>
+    <div class="crayons-modal__box__body flag-user-modal-body">
+      <span>
+        Thanks for keeping DEV safe. Here is what you can do to flag this user:
+      </span>
+      <div class="crayons-fields">
+        <div class="crayons-field crayons-field--radio">
+          <input
+            type="radio"
+            id="vomit-all"
+            name="flag-user"
+            class="crayons-radio"
+            data-reactable-id="${articleAuthorId}"
+            data-category="vomit"
+            data-reactable-type="${articleAuthorClassName}">
+          <label for="vomit-all" class="crayons-field__label">
+            Make all posts by this author less visible
+            <p class="crayons-field__description">
+              This author consistently posts content that violates DEV's code of conduct because it is harassing, offensive or spammy.
+            </p>
+          </label>
+        </div>
+        <a href="/report-abuse?url=${document.location}" class="fs-base abuse-report-link">Report other inappropriate conduct</a>
+      </div>
+      <div class="buttons-container">
+        <a href="#" class="crayons-btn crayons-btn--danger" id="confirm-flag-user-action">Confirm action</a>
+        <a href="#" class="crayons-btn crayons-btn--secondary" id="cancel-flag-user-action">Cancel</a>
+      </div>
+    </div>
+  </div>
+  <div class="crayons-modal__overlay"></div>
+</div>
+`;
+
+  if (user.id !== articleAuthorId && user.trusted) {
+    const modContainer = document.getElementById('mod-container');
+
+    document.querySelector(
+      '.modal-snackbar-container',
+    ).innerHTML = modalSnackbarHTML;
+    document.querySelector(
+      '.flag-user-modal-container',
+    ).innerHTML = flagUserModalHTML;
+    modContainer.addEventListener('load', () => {
+      modContainer.contentWindow.document
+        .getElementById('open-flag-user-modal')
+        .addEventListener('click', toggleFlagUserModal);
+    });
+  }
+
+  // Event listeners to Close the Modal
+  const closeModalElements = Array.from(
+    document.querySelectorAll(
+      '.crayons-modal__overlay, .modal-header-close-icon, #cancel-flag-user-action',
+    ),
+  );
+
+  closeModalElements.forEach((element) => {
+    element.addEventListener('click', toggleFlagUserModal);
+  });
+
+  document
+    .getElementById('confirm-flag-user-action')
+    .addEventListener('click', () => {
+      if (!document.getElementById('vomit-all').checked) {
+        alert('No Selection Made!');
+      } else {
+        toggleFlagUserModal();
+      }
+    });
+
+  document
+    .getElementById('confirm-flag-user-action')
+    .addEventListener('click', (e) => {
+      e.preventDefault();
+      const vomitAllOption = document.getElementById('vomit-all');
```

Comment:
You can use the [:checked(https://developer.mozilla.org/en-US/docs/Web/CSS/:checked) pseudo class.
```suggestion
      const vomitAllOption = document.querySelector('vomit-all:checked');
```

Reply:
Thank you!